Update the docs to use the new "runner" classes and terminology. (#969)
sematic-ai/sematic: docs/glossary.md

describe pipelines.

What we call a "pipeline" is typically the root Sematic Function. It is the
Sematic Function on which you called `runner.run(some_sematic_function())`.

See [Root function](#root-entry-point-function).

## Runners

{% hint style="info" %}

> This concept used to be referred to as Resolvers. So don't

Reviewer: ```suggestion
This concept used to be referred to as `Resolvers`. So don't
```

worry if you're familiar with that terminology! Everything
you know about Resolvers applies to Runners as well, except

> that `.resolve(...)` has been renamed to `.run(...)`. 

Reviewer: Here as well.

{% endhint %}

> Runners dictate how your pipeline gets "ran". Running a pipeline means

Reviewer: ```suggestion
Runners dictate how your pipeline get executed. Running a pipeline means
```

Author: Changed to "run" as I believe it is more grammatically valid. But I think the idea here is to maintain parallelism between the words used for "Runner" and the action they perform. That parallelism was present in the original "Resolver"/"gets resolved" structure of these docs, and I am trying to preserve it here.

going through its DAG (in-memory graph of `Future` objects), and proceeding to
executing each step as its inputs are available.

Different runners offer different execution strategies:

- **`SilentRunner`** – will run a pipeline without persisting anything to
  disk or the database. This is ideal for testing or iterating without poluting
  the database. This also means that pipelines ran with `SilentRunner`
  are not tracked and therefore not visualizable in the web dashboard.
- **`LocalRunner`** – will run a pipeline on the machine where it was
  called (typically you local dev machine). It will persist artifacts and track
  metadata in the database. Runs will be visualizable in the dashboard. No
  parallelization is applied, the graph is topologically sorted. See [Local
  execution](./local-execution.md).
- **`CloudRunner`** – will submit a pipeline to execute on a Kubernetes
  cluster. This can be used to leverage step-dependent cloud resources (e.g.
  GPUs, high-memory VMs, etc.). See [Cloud runner](./cloud-runner.md).

## Pipeline Run

{% hint style="info" %}

> This concept used to be referred to as "Resolution". So don't

Reviewer: ```suggestion
This concept used to be referred to as `Resolution`. So don't
```

Author: done

worry if you're familiar with that terminology! Everything

> you know about Resolutions applies to Pipeline Runs as well.

Author: done
